tools/replay/parse_dump_resources_cli.cpp

```diff
@@ -437,19 +445,31 @@ bool parse_dump_resources_arg(gfxrecon::decode::VulkanReplayOptions& vulkan_repl
                     vulkan_replay_options.RenderPass_Indices[i][0].push_back(EndRenderPass);
                 }
 
+                vulkan_replay_options.Dispatch_Indices.push_back(std::vector<uint64_t>());
                 if (Dispatch)
                 {
-                    vulkan_replay_options.Dispatch_Indices.push_back(std::vector<uint64_t>());
                     vulkan_replay_options.Dispatch_Indices[i].push_back(Dispatch);
                 }
 
+                vulkan_replay_options.TraceRays_Indices.push_back(std::vector<uint64_t>());
                 if (TraceRays)
                 {
-                    vulkan_replay_options.TraceRays_Indices.push_back(std::vector<uint64_t>());
                     vulkan_replay_options.TraceRays_Indices[i].push_back(TraceRays);
                 }
 
-                vulkan_replay_options.QueueSubmit_Indices.insert(QueueSubmit);
+                if (vulkan_replay_options.QueueSubmit_Indices.find(QueueSubmit) !=
```

Comment:
Could you use the idiom here `auto [itr, inserted] = vulkan_replay_options.QueueSubmit_Indices.insert(QueueSubmit);` and then `if(!inserted) {parse_error = true...}`?  `std::unordered_set::insert` will fail if the element is already in there.  If you'd like I'll make the commit but I don't have a test case here.

Reply:
Change made in 15d6ab7561eccb3b1c1c40c0f9c8b412bc5ba741